runtime/common/src/lib.rs

```diff
@@ -121,37 +121,48 @@ pub mod types {
 
 	/// A representation of a tranche weight, used to weight
 	/// importance of a tranche
+	#[derive(Encode, Decode, Copy, Debug, Default, Clone, PartialEq, Eq, TypeInfo, CompactAs)]
+	#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+	pub struct TrancheWeight(pub u128);
+
+	/// A representation of InstanceId for Uniques.
 	#[derive(
-		codec::Encode,
-		codec::Decode,
-		Copy,
-		Debug,
+		Encode,
+		Decode,
 		Default,
+		Copy,
 		Clone,
 		PartialEq,
 		Eq,
+		CompactAs,
+		Debug,
+		MaxEncodedLen,
 		TypeInfo,
-		codec::CompactAs,
 	)]
 	#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
-	pub struct TrancheWeight(pub u128);
+	pub struct InstanceId(pub u128);
 
-	/// A representation of InstanceId for Uniques.
+	/// A type describing the additional metadata stored in the OrmlAssetRegistry
 	#[derive(
-		codec::Encode,
-		codec::Decode,
-		Default,
-		Copy,
 		Clone,
+		Copy,
+		Default,
+		PartialOrd,
+		Ord,
 		PartialEq,
 		Eq,
-		codec::CompactAs,
 		Debug,
-		codec::MaxEncodedLen,
+		Encode,
+		Decode,
 		TypeInfo,
+		MaxEncodedLen,
 	)]
-	#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
-	pub struct InstanceId(pub u128);
+	pub struct CustomMetadata {
+		pub fee_per_second: Option<Balance>,
```

Comment:
Can you leave a comment what `fee_per_second` here means? I know it's XCM related, but that's not entirely clear in the context here.

Reply:
good point! I could also have a xcm-related-fields struct to offer context and the type-level? Or would a comment be enough for you? I didn't go with the former because there is only one field now and I don't foresee more for now.